packages/entities/entities-plugins/src/components/PluginEntityForm.vue

```diff
@@ -0,0 +1,717 @@
+    })
+  }
+
+  return axiosInstance.get(url, { params })
+}
+
+// provide to KFG
+provide(FORMS_API_KEY, {
+  getOne,
+  getAll,
+})
+
+const sharedFormName = ref('')
+const form = ref<Record<string, any> | null>(null)
+const formSchema = ref<Record<string, any>>({})
+const originalModel = reactive<Record<string, any>>({})
+const formModel = reactive<Record<string, any>>({})
+const formOptions = computed(() => form.value?.options)
+
+const entityIdField = computed((): string => {
+  switch (props.entityType) {
+    case 'service':
+      return EntityTypeIdField.SERVICE
+    case 'route':
+      return EntityTypeIdField.ROUTE
+    case 'consumer':
+      return EntityTypeIdField.CONSUMER
+    case 'consumer_group':
+      return EntityTypeIdField.CONSUMER_GROUP
+    default:
+      return ''
+  }
+})
+
+const onModelUpdated = (model: Record<string, any>, schema: string) => {
+  const newData = { [schema]: model }
+  const newModel = Object.assign({}, formModel, newData)
+
+  Object.assign(formModel, newModel)
+
+  emit('model-updated', {
+    model: formModel,
+    originalModel,
+    data: getModel(),
+  })
+}
+
+const updateModel = (data: Record<string, any>, parent?: string) => {
+  Object.keys(data).forEach(key => {
+    let modelKey = parent ? `${parent}-${key}` : key
+    let scheme = props.schema[modelKey]
+
+    // If `scheme` is undefined, it is either because the key is not the deepest nested key of the schema object
+    // or the field name has dashes in it and its schema key was converted to underscores during initiation.
+    if (!scheme) {
+      const underscoredModelKey = parent ? `${parent}-${key.replace(/-/g, '_')}` : key.replace(/-/g, '_')
+
+      // Here we check if the underscored key exists in the schema and the `fieldNameHasDashes` flag is true
+      if (props.schema[underscoredModelKey]?.fieldNameHasDashes) {
+        modelKey = underscoredModelKey
+        scheme = props.schema[modelKey]
+      }
+    }
+
+    const value = data[key]
+    const type = typeof value
+
+    // Ensure Object Advanced field is saved to model as empty object
+    if (scheme && scheme.type === 'object-advanced' && !value) {
+      formModel[modelKey] = {}
+      originalModel[modelKey] = {}
+
+      return
+    }
+
+    if (Array.isArray(value)) {
+      formModel[modelKey] = JSON.parse(JSON.stringify(value))
+      originalModel[modelKey] = JSON.parse(JSON.stringify(value))
+
+      return
+    }
+
+    if (type === 'object' && value && !value.length) {
+      return updateModel(value, modelKey)
+    }
+
+    if (type === 'object' && value && value.length && scheme.type === 'input') {
+      formModel[modelKey] = value.join(', ')
+      originalModel[modelKey] = value.join(', ')
+
+      return
+    }
+
+    formModel[modelKey] = value
+    originalModel[modelKey] = value
+  })
+
+  emit('model-updated', {
+    model: formModel,
+    originalModel,
+    data: getModel(),
+  })
+}
+
+const getModel = (): Record<string, any> => {
```

Comment:
nit: you're calling this function above before it is defined; you should move the function definition above the `updateModel` block